<!--
  1 failure:  :disappointed_rel...
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.5.0 test /Users/build/jenkins/workspace/dk_titanium_mobile_PR-10310-XR5OTGJO7ZSDDI7ENHTV2WRHGLBYMFWAD2Y7SPPZIIHQSTXUOJDA
> grunt

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/dk_titanium_mobile_PR-10310-XR5OTGJO7ZSDDI7ENHTV2WRHGLBYMFWAD2Y7SPPZIIHQSTXUOJDA/iphone/cli/commands/_build.js
  202:114  warning  'arr' is defined but never used  no-unused-vars

✖ 1 problem (0 errors, 1 warning)


Running "checkFormat:ios" (checkFormat) task

Running "checkFormat:android" (checkFormat) task

Running "validate:docs" task
Warning: Docs validation failed with exit code: 1
Output: [ERROR] ./Titanium/Proxy.yml
Titanium.Proxy: found 1 error(s)!
	methods
		applyProperties
			parameters
				props
					Invalid key(s) in Titanium.Proxy: since

[ERROR] Found 1 error(s)!
 Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

--- a/apidoc/Titanium/UI/Tab.yml
+++ b/apidoc/Titanium/UI/Tab.yml
@@ -306,7 +306,7 @@ events:
 
   - name: unselected
     summary: Fired when the tab is no longer selected.
-    since: {android: "3.5.1 ",  iphone: "5.2.0", ipad: "5.2.0"}
+    since: {android: "3.5.1",  iphone: "5.2.0", ipad: "5.2.0"}
     properties:
       - name: index
         summary: Index of the current active tab.
@@ -348,7 +348,7 @@ events:
 
   - name: selected
     summary: Fired when the tab is selected.
-    since: {android: "3.5.1 ",  iphone: "5.2.0", ipad: "5.2.0"}
+    since: {android: "3.5.1",  iphone: "5.2.0", ipad: "5.2.0"}
     properties:
       - name: index
         summary: Index of the current active tab.

--- a/apidoc/Titanium/UI/View.yml
+++ b/apidoc/Titanium/UI/View.yml
@@ -268,7 +268,7 @@ events:
         fired when the property [focusable](Titanium.UI.View.focusable) is set to true. On iOS the
         event is generated only when using [Ti.UI.TextArea](Titanium.UI.TextArea), [Ti.UI.TextField](Titanium.UI.TextField)
         and [Ti.UI.SearchBar](Titanium.UI.SearchBar).
-    platforms: [android]
+    platforms: [android, iphone, ipad]
     since: {android: "3.1.0", iphone: "5.4.0", ipad: "5.4.0"}
     properties:
       - name: keyCode

--- a/apidoc/docgen.js
+++ b/apidoc/docgen.js
@@ -576,9 +576,7 @@ function addOnMerge(baseObj, addObj) {
 					baseObj[key] = addOnMerge(base, add);
 					break;
 				case 'string':
-					if (!~[ 'name', 'since', '__file' ].indexOf(key)) {
-						baseObj[key] += ' ' + add;
-					} else if (key === 'since') {
+					if (key === 'since') {
 						const platforms = baseObj.platforms || Object.keys(common.DEFAULT_VERSIONS);
 						const since = {};
 
@@ -598,7 +596,6 @@ function addOnMerge(baseObj, addObj) {
 							common.log(common.LOG_WARN, 'Cannot set since version.  Set since as a dictionary or add the platforms property.');
 							break;
 						}
-
 						baseObj[key] = since;
 					}
 					break;

--- a/apidoc/lib/changes_generator.js
+++ b/apidoc/lib/changes_generator.js
@@ -77,15 +77,39 @@ function checkVersions(api, startVersion, endVersion, className) {
 
 	if (assert(api, 'deprecated')) {
 		if (assert(api.deprecated, 'removed')) {
-			if (nodeappc.lte(startVersion, api.deprecated.removed)) {
-				if (!endVersion || (endVersion && nodeappc.gt(endVersion, api.deprecated.removed))) {
-					rv.removed = true;
+			let versions = [];
+			if (api.deprecated.removed instanceof Object) {
+				for (let key in api.deprecated.removed) {
+					versions.push(api.deprecated.removed[key]);
+				}
+			} else {
+				versions.push(api.deprecated.removed);
+			}
+			for (let version of versions) {
+				if (nodeappc.lte(startVersion, version)) {
+					if (!endVersion || (endVersion && nodeappc.gt(endVersion, version))) {
+						rv.removed = true;
+						break;
+					}
 				}
 			}
 		}
-		if (nodeappc.lte(startVersion, api.deprecated.since)) {
-			if (!endVersion || (endVersion && nodeappc.gt(endVersion, api.deprecated.since))) {
-				rv.deprecated = true;
+		if (assert(api.deprecated, 'since')) {
+			let versions = [];
+			if (api.deprecated.since instanceof Object) {
+				for (let key in api.deprecated.since) {
+					versions.push(api.deprecated.since[key]);
+				}
+			} else {
+				versions.push(api.deprecated.since);
+			}
+			for (let version of versions) {
+				if (nodeappc.lte(startVersion, version)) {
+					if (!endVersion || (endVersion && nodeappc.gt(endVersion, version))) {
+						rv.deprecated = true;
+						break;
+					}
+				}
 			}
 		}
 		rv.name = className ? className + '.' + api.name : api.name;

